### PR TITLE
Add or improve library doc [ci: last-only]

### LINKED ISSUE #12900: Improve documentation for deprecated methods that only make sense for parallel collections
at https://github.com/scala/toolkit/issues/31#issuecomment-1794225687 @philipschwarz noted:

> Maybe instead of just saying that the aggregate function 
> is not relevant for sequential collections. Use `foldLeft(z)(seqop) instead., "2.13.0
> the [Scala Doc for aggregate]> (https://github.com/scala/scala/blob/v2.13.10/src/library/scala/collection/IterableOnce.scala#L1122) could also say that aggregate is relevant for parallel collections and it could provide a link to https://github.com/scala/scala-parallel-collections.

perhaps the method should also be deprecated? and perhaps there are other such methods? @scala/collections

### LINKED ISSUE #12904: Missing doc comments for collection operations
## Reproduction steps

Looking at

[ScalaDoc for ListBuffer](https://www.scala-lang.org/api/2.13.6/scala/collection/mutable/ListBuffer.html)

I note that there are quite a few operations that are public but that have no doc comment whatsoever! For instance, `dropInPlace`, `mapInPlace`, `prependAll`. 

I consider that a bug. Every collection operation should be documented. This bug is probably worse than a failure of type inference
or a compiler crash in obscure cases since it gives a shoddy first impression.

### PR DESCRIPTION
**Pull request by som-snytt**
Supply some missing doc for some public API, and that part fixes scala/bug#12904. Also fixes https://github.com/scala/bug/issues/12900.

Add an internal-only option to scaladoc to detect missing doc, but requires refinement not to be noisy on stdlib, even restricted to collections.

### CODE REVIEW

**Comment by som-snytt** [comment]
`throws` requires text, e.g., `PriorityQueue#dequeue`. Otherwise it is silently ignored (same for dottydoc).
```
   *  @throws NoSuchElementException if no element to remove from heap
```

**Comment by som-snytt** [comment]
@SethTisue "it's only doc". I will follow up right now with improvements to `aggregate` as lauded on the linked ticket.

That might not seem like a big deal, but according to wikipedia,

> In 2017, the aggregate industry in the United States was valued at US$20.9 billion.

File: src/scaladoc/scala/tools/nsc/doc/html/page/Entity.scala [hunk 1]
```diff
@@ -384,16 +384,19 @@ trait EntityPage extends HtmlPage {
...
     mbr match {
       // comment of class itself
       case dte: DocTemplateEntity if isSelf =>
         Div(id="comment", `class`="fullcommenttop", elems= memberToCommentBodyHtml(dte, inTpl, isSelf = true))
       case _ =>
         // comment of non-class member or non-documented inner class
         val commentBody = memberToCommentBodyHtml(mbr, inTpl, isSelf = false)
-        if (commentBody.isEmpty)
+        if (commentBody.isEmpty) {
+          if (universe.settings.docRequired.value && mbr.visibility.isPublic && inTpl.visibility.isPublic && mbr.toString.startsWith("scala.collection") && mbr.isInstanceOf[Def])
```
Comment (lrytz) [inline src/scaladoc/scala/tools/nsc/doc/html/page/Entity.scala:396]: Maybe a make this generally available as `-required:package.prefix`?

**Review by lrytz — COMMENTED** [review]
LGTM otherwise, thanks a lot for the docwork!

**Comment by som-snytt** [comment]
I restricted this PR to documentation only. Hopefully we can merge it. I just spent time trying to find my doc for `aggregate`.

The scaladoc tool flag isInternalOnly, contingent upon further refinement.

File: src/scaladoc/scala/tools/nsc/doc/html/page/Entity.scala (hunk 1 above)
Comment (som-snytt) [reply src/scaladoc/scala/tools/nsc/doc/html/page/Entity.scala:396]: I made it less generally available for purposes of this PR.

**Pull request merged**